src/jit/lower.cpp

```diff
@@ -432,17 +432,34 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
             // relink them into the proper order, i.e. loOp1 is followed by loOp2, and then
```

Comment:
It would be better to refactor this code out into it's own method as it is now around 50 lines of  complex code.
This could be done in a future changeset.